### PR TITLE
feat(atomic): add new atomic-commerce-refine-toggle component

### PR DESCRIPTION
**Pull request by olamothe**
Add support for `atomic-commerce-refine-toggle` (display sort and facet in mobile)

image

image

* New "internal" `atomic-commerce-refine-modal` which is in charge of displaying sorts and filters in a modal when the toggle button is displayed
* Refactor `atomic-refine-toggle` and `atomic-refine-modal` to extract common patterns
* Small refactor to `layout` components to take into account the different component tag name (ie: `atomic-refine-toggle` vs `atomic-commerce-refine-toggle`
* Bugfixes in Headless for Breadcrumb (both commerce and search use cases) related to the `hasBreadcrumb` property being incorrect
* 

https://coveord.atlassian.net/browse/KIT-3330

### CODE REVIEW

File: packages/atomic/src/components/commerce/facets/atomic-commerce-facet/atomic-commerce-facet.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ import {
   FacetValue,
 } from '../../../common/facets/facet-value/facet-value';
 import {FacetValuesGroup} from '../../../common/facets/facet-values-group/facet-values-group';
-import {initializePopover} from '../../../search/facets/atomic-popover/popover-type';
+import {initializePopover} from '../../../common/facets/popover/popover-type';
```
Comment (alexprudhomme) [inline packages/atomic/src/components/commerce/facets/atomic-commerce-facet/atomic-commerce-facet.tsx:45]: Is the atomic-popover even usable in commerce ? How do you use it ?

**Review by alexprudhomme — APPROVED** [review]
lgtm

**Review by pathiery — APPROVED** [review]

File: packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.test.ts [hunk 2]
```diff
@@ -564,11 +564,79 @@ describe('headless breadcrumb manager', () => {
...
+            values: [],
+          }),
+        }),
+      },
+    };
+
+    expect(breadcrumbManager.state.hasBreadcrumbs).toBe(false);
+  });
+
+  it('hasBreadcrumbs returns false when an automatic facet has idle value', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.test.ts:623]:
```suggestion
  it('hasBreadcrumbs returns false when an automatic facet only has idle values', () => {
```

File: packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.test.ts [hunk 3]
```diff
@@ -108,21 +109,50 @@ describe('core breadcrumb manager', () => {
...
+        buildMockCommerceRegularFacetResponse({
+          facetId,
+          values: [buildMockCommerceRegularFacetValue({state: 'selected'})],
+        })
+      );
+
+      expect(breadcrumbManager.state.hasBreadcrumbs).toEqual(true);
+    });
+
+    it('#hasBreadcrumbs is true when there is at least one excluded facet value', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.test.ts:146]:
```suggestion
    it('#hasBreadcrumbs is true when there is an excluded facet value', () => {
```

File: packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.test.ts [hunk 4]
```diff
@@ -108,21 +109,50 @@ describe('core breadcrumb manager', () => {
...
+        buildMockCommerceRegularFacetResponse({
+          facetId,
+          values: [buildMockCommerceRegularFacetValue({state: 'idle'})],
+        })
+      );
+
+      expect(breadcrumbManager.state.hasBreadcrumbs).toEqual(false);
+    });
+
+    it('#hasBreadcrumbs is true when there is at least one selected facet value', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.test.ts:135]:
```suggestion
    it('#hasBreadcrumbs is true when there is a selected facet value', () => {
```

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by Spuffynism — APPROVED** [review]
Commerce part lgtm!

**Review by dmgauthier — APPROVED** [review]

**Pull request merged**